scipy/signal/tests/test_filter_design.py

```diff
@@ -3734,3 +3734,76 @@ def test_w_or_N_types(self):
+    # Test erroneus input cases.
+    def test_invalid_input(self):
+        # Cutoff frequency is <= 0 or >= fs / 2.
+        fs = 16000
+        with pytest.raises(ValueError, match='The frequency must be between '):
+            gammatone(-fs, 'iir', fs=fs)
+            gammatone(0, 'fir', fs=fs)
+            gammatone(fs / 2, 'iir', fs=fs)
+            gammatone(fs, 'fir', fs=fs)
+
+        # Filter type is not fir or iir
+        with pytest.raises(ValueError, match='ftype must be '):
+            gammatone(440, ftype='fie', fs=16000)
+            gammatone(220, ftype='it', fs=32000)
+
+        # Order is <= 0 or > 24 for FIR filter.
+        with pytest.raises(ValueError, match='Invalid order: '):
+            gammatone(440, 'fir', fs=16000, order=-50)
+            gammatone(220, 'fir', fs=32000, order=0)
+            gammatone(110, 'fir', fs=44100, order=25)
+            gammatone(55, 'fir', fs=48000, order=50)
+
+    # Verify that the filter's frequency response is approximately
+    # 1 at the cutoff frequency.
+    def test_frequency_response(self):
+        fs = 16000
+        ftypes = ['fir', 'iir']
+        for ftype in ftypes:
+            # Create a gammatone filter centered at 1000 Hz.
+            b, a = gammatone(1000, ftype, fs=fs)
+
+            # Calculate the frequency response.
+            freqs, response = freqz(b, a)
+
+            # Determine peak magnitude of the response
+            # and corresponding frequency.
+            response_max = np.max(np.abs(response))
+            freq_hz = freqs[np.argmax(np.abs(response))] / ((2 * np.pi) / fs)
+
+            # Check that the peak magnitude is 1 and the frequency is 1000 Hz.
+            response_max == pytest.approx(1, rel=1e-2)
+            freq_hz == pytest.approx(1000, rel=1e-2)
+
+    # All built-in IIR filters are real, so should have perfectly
+    # symmetrical poles and zeros. Then ba representation (using
+    # numpy.poly) will be purely real instead of having negligible
+    # imaginary parts.
+    def test_iir_symmetry(self):
+        b, a = gammatone(440, 'iir', fs=24000)
+        z, p, k = tf2zpk(b, a)
+        sorted(z) == pytest.approx(sorted(z.conj()))
+        sorted(p) == pytest.approx(sorted(p.conj()))
+        k == pytest.approx(np.real(k))
+
+        assert_(issubclass(b.dtype.type, np.floating))
+        assert_(issubclass(a.dtype.type, np.floating))
+
+    # Verify IIR filter coefficients with MATLAB's answers
+    def test_iir_ba_output(self):
+        b, a = gammatone(440, 'iir', fs=16000)
+        b2 = [1.31494461367464e-06, -5.03391196645395e-06,
+              7.00649426000897e-06, -4.18951968419854e-06,
+              9.02614910412011e-07]
+        a2 = [1.0, -7.65646235454218,
+              25.7584699322366, -49.7319214483238,
+              60.2667361289181, -46.9399590980486,
+              22.9474798808461, -6.43799381299034,
+              0.793651554625368]
+        b == pytest.approx(b2)
+        a == pytest.approx(a2)
```

Comment:
In SciPy it's more standard to use `assert_allclose` because it gives precise control over absolute and relative tolerances. So probably better to use it here rather than introduce `pytest.approx`

Reply:
Fixed. I kept `pytest.approx` on the frequency response unit test because it's just comparing one number and it's faster than `assert_approx_equal`.